### DISCUSSION TITLE
Resolve openid.issuer filed dynamically

### DISCUSSION THREAD

**Discussion by jajaros** [micronaut-projects/micronaut-security#824] [Q&A] [answered]
Hello,
I have an application serving multiple domains, and for each of them, I want to use different openid.issuer.
My goal is to provide _OauthClientConfigurationProperties.OpenIdClientConfigurationProperties#issuer_ dynamically, from a _@RequestScope_ bean.
The field is, however used during the application startup and such bean cannot be created at that time.
So my I'm wondering - is there any way how to achieve this?

Thanks for any suggestions.

**Comment by sdelamo** [comment, marked as answer]
We don't have a clear solution for your use case. You probably should don't try to create `OauthClientConfigurationProperties.OpenIdClientConfigurationProperties` dynamically but you will have a to inviestigate about creating per request a `OpenIdClientConfiguration` and also check `OpenIdClientFactory`. 

Do you know all the domains of your application? Maybe you could add an oauth client per domain.

**Reply by jajaros** [reply]
We ended up implementing own `OpenIdClientFactory `and `OpenIdClient` which resolves it per request and it seems to work.
There will be dozens of domains, so it's not feasible to have a client per domain.

Is there any chance this will be somehow addressed in the future versions of Micronaut?
Thanks!

**Reply by sdelamo** [reply]
Happy, you managed to work around it.

> Is there any chance this will be somehow addressed in the future versions of Micronaut?

not sure. I am not sure your usecase is that common. 

 it maybe useful for you to create an issue describing what you did and what it achieves to evaIuate for the future
